jina/serve/runtimes/gateway/http/app.py

```diff
@@ -165,11 +168,39 @@ async def post(body: JinaEndpointRequestModel):
             if bd['data'] is not None and 'docs' in bd['data']:
                 req_generator_input['data'] = req_generator_input['data']['docs']
 
-            result = await _get_singleton_result(
-                request_generator(**req_generator_input)
-            )
+            try:
+                result = await _get_singleton_result(
+                    request_generator(**req_generator_input)
+                )
+            except InternalNetworkError as err:
+                response.status_code = status.HTTP_503_SERVICE_UNAVAILABLE
+                result = bd  # send back the request
+                result['header'] = _generate_exception_header(
+                    err
+                )  # attach exception details to response header
+                logger.error(
+                    f'Error while getting responses from deployments: {err.details()}'
+                )
             return result
 
+    def _generate_exception_header(error: InternalNetworkError):
+        import traceback
+
+        exception_dict = {
+            'name': str(error.__class__),
+            'stacks': [
+                str(x) for x in traceback.extract_tb(error.og_exception.__traceback__)
+            ],
+            'executor': '',
+        }
+        status_dict = {
+            'code': 3,  # status error
```

Comment:
where can I see what each code means?

Reply:
They are defined [here](https://github.com/jina-ai/jina/blob/master/jina/proto/jina.proto#L57). I don't like setting it manually like this either, but couldn't see a better way.